### PR TITLE
common: add message to deprecated macros

### PR DESCRIPTION
**Pull request by edoardolombardi**
Added new macro to define functions as deprecated with message as input argument.

With this new macros BITPIT_DEPRECATED_FOR (one for each supported compiler as the original ones) is possibile to pass a message to indicate wich method has to be used instead.

Currently compiled and tested only for GNUC case.

### CODE REVIEW

File: src/common/compiler.hpp [hunk 1]
```diff
@@ -75,10 +75,13 @@ do {                  \
  */
 #if defined __GNUC__
 #   define BITPIT_DEPRECATED(func) func __attribute__ ((deprecated))
+#   define BITPIT_DEPRECATED_FOR(func, replacement) func __attribute__ ((deprecated(" Use " #replacement)))
 #elif defined __clang__
 #   define BITPIT_DEPRECATED(func) func __attribute__ ((deprecated))
+#   define BITPIT_DEPRECATED(func, replacement) func __attribute__ ((deprecated(" Use " #replacement)))
```
Comment (andrea-iob) [inline src/common/compiler.hpp:81]: Maybe this should be "BITPIT_DEPRECATED_FOR"?

File: src/common/compiler.hpp [hunk 2]
```diff
@@ -75,10 +75,13 @@ do {                  \
...
 #elif defined(_MSC_VER)
     #define BITPIT_DEPRECATED(func) __declspec(deprecated) func
+    #define BITPIT_DEPRECATED(func), replacement __declspec(deprecated(" Use " #replacement)) func
```
Comment (andrea-iob) [inline src/common/compiler.hpp:84]: Same here.

Comment (edoardolombardi) [reply src/common/compiler.hpp:84]: Sure, thanks. I put two bugs in three lines of code... 🗡️

**Comment by andrea-iob** [comment]
Could you please also add "define BITPIT_DEPRECATED_FOR(func, replacement) func" in the else branch? This will allow unsupported compilers to still compile bitpit even if we don't explicitly define a deprecated macro for them.

**Review by andrea-iob — APPROVED** [review]

**Comment by andrea-iob** [comment]
Since we are changing this part of the code, I changed the message displayed when deprecated macros are not implemented ad fixed the indentation.

**Pull request merged**